src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -611,84 +602,94 @@ void BasePlaylistFeature::slotExportPlaylist() {
 
 // MV added method
 void BasePlaylistFeature::slotExportPlaylists() {
-    int playlistId = playlistIdFromIndex(m_lastRightClickedIndex);
-    if (playlistId == kInvalidPlaylistId) {
-        return;
-    }
-    QString playlistName = m_playlistDao.getPlaylistName(playlistId);
-    // replace separator character with something generic
-    playlistName = playlistName.replace(QDir::separator(), kUnsafeFilenameReplacement);
-    qDebug() << "Export playlist" << playlistName;
+    const PlaylistDAO::HiddenType hidden = PlaylistDAO::PLHT_NOT_HIDDEN;
+    QList<QPair<int, QString>> test2 = m_playlistDao.getPlaylists(hidden);
 
-    QString lastPlaylistDirectory = m_pConfig->getValue(
-            kConfigKeyLastImportExportPlaylistDirectory,
-            QStandardPaths::writableLocation(QStandardPaths::MusicLocation));
+    QList<QPair<int, QString>> allPlaylists = m_playlistDao.getPlaylists(hidden);
 
-    // Open a dialog to let the user choose the file location for playlist export.
-    // The location is set to the last used directory for import/export and the file
-    // name to the playlist name.
-    const QString fileLocation = getFilePathWithVerifiedExtensionFromFileDialog(
-            tr("Export Playlist"),
-            lastPlaylistDirectory.append("/").append(playlistName).append(".m3u"),
-            tr("M3U Playlist (*.m3u);;M3U8 Playlist (*.m3u8);;"
-               "PLS Playlist (*.pls);;Text CSV (*.csv);;Readable Text (*.txt)"),
-            tr("M3U Playlist (*.m3u)"));
-    // Exit method if the file name is empty because the user cancelled the save dialog.
-    if (fileLocation.isEmpty()) {
-        return;
-    }
-
-    // Update the import/export playlist directory
-    QString fileDirectory(fileLocation);
-    fileDirectory.truncate(fileLocation.lastIndexOf("/"));
-    m_pConfig->set(kConfigKeyLastImportExportPlaylistDirectory,
-            ConfigValue(fileDirectory));
-
-    // The user has picked a new directory via a file dialog. This means the
-    // system sandboxer (if we are sandboxed) has granted us permission to this
-    // folder. We don't need access to this file on a regular basis so we do not
-    // register a security bookmark.
+    for (int i = 0; i < allPlaylists.length(); i++) {
+        QPair thePair = allPlaylists.at(i);
+        int playlistId = thePair.first;
+        QString playlistName = m_playlistDao.getPlaylistName(playlistId);
```

Comment:
This additional lookup _per playlist_ is redundant, you already get the name as `thePair.second`.
some more meaningful names would are always welcome, `thePair` > `idName`?

Reply:
Solved by changing that loop to range based loop: 
```cpp
    for (const auto& [playlistId, playlistName] : allPlaylists) {
        ...
    }

```